orbit/pkg/table/diskutil/apfs/apfs_darwin.go

```diff
@@ -0,0 +1,179 @@
+				"role":                                role,
+				"capacity_in_use":                     strconv.FormatInt(volume.CapacityInUse, 10),
+				"capacity_quota":                      strconv.FormatInt(volume.CapacityQuota, 10),
+				"capacity_reserve":                    strconv.FormatInt(volume.CapacityReserve, 10),
+				"crypto_migration_on":                 convertBool(volume.CryptoMigrationOn),
+				"encryption":                          convertBool(volume.Encryption),
+				"filevault":                           convertBool(volume.FileVault),
+				"locked":                              convertBool(volume.Locked),
+			})
+		}
+	}
+	return rows, nil
+}
+
+// Columns is the schema of the table.
+func PhysicalStoresColumns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		table.TextColumn("container_uuid"),
+		table.TextColumn("container_designated_physical_store"),
+		table.TextColumn("container_reference"),
+		table.IntegerColumn("container_fusion"),
+		table.BigIntColumn("container_capacity_ceiling"),
+		table.BigIntColumn("container_capacity_free"),
+		table.TextColumn("uuid"),
+		table.TextColumn("identifier"),
+		table.BigIntColumn("size"),
+	}
+}
+
+// Generate is called to return the results for the table at query time.
+// Constraints for generating can be retrieved from the queryContext.
+func PhysicalStoresGenerate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	cmd := exec.Command("/usr/sbin/diskutil", "apfs", "list", "-plist")
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("generate failed: %w", err)
+	}
+
+	cmdResult, err := parseReadOutput(out)
+	if err != nil {
+		return nil, fmt.Errorf("parse authorizationdb read output: %w", err)
+	}
+
+	rows := make([]map[string]string, 0)
+
+	for _, container := range cmdResult.Containers {
+		for _, physicalStore := range container.PhysicalStores {
+			rows = append(rows, map[string]string{
+				"container_uuid":                      container.APFSContainerUUID,
+				"container_designated_physical_store": container.DesignatedPhysicalStore,
+				"container_reference":                 container.ContainerReference,
+				"container_fusion":                    convertBool(container.Fusion),
+				"container_capacity_ceiling":          strconv.FormatInt(container.CapacityCeiling, 10),
+				"container_capacity_free":             strconv.FormatInt(container.CapacityFree, 10),
+				"uuid":                                physicalStore.DiskUUID,
+				"identifier":                          physicalStore.DeviceIdentifier,
+				"size":                                strconv.FormatInt(physicalStore.Size, 10),
+			})
+		}
+	}
+	return rows, nil
+}
+
+func parseReadOutput(out []byte) (*CmdResult, error) {
+	var m CmdResult
+	if _, err := plist.Unmarshal(out, &m); err != nil {
+		return nil, err
+	}
+	return &m, nil
+}
+
+func convertBool(b bool) string {
+	if b {
+		return "1"
+	}
+	return "0"
+}
```

Comment:
We should add unit tests with a sample output, e.g. https://github.com/fleetdm/fleet/blob/main/orbit/pkg/table/pmset/pmset_darwin_test.go

Reply:
I added some tests for the entire table generation, this should be done